File: aes_kurrr/generate.php
<?php

$message = "Hello, this is a secret message! Don't tell anyone that you got it, You must help Me. Right now I'm sitting in the prison"; // Повідомлення для шифрування

// Генерація випадкового 128-бітного ключа (16 байтів)
$key = openssl_random_pseudo_bytes(16);
echo "Generated Key (hex): " . bin2hex($key) . "\n";

// Збереження ключа у файл
file_put_contents('aes_key.bin', $key);

// Ініціалізація шифру AES у режимі ECB
$encryptedMessage = openssl_encrypt($message, 'AES-128-ECB', $key, OPENSSL_RAW_DATA);

// Перевірка успішності шифрування
if ($encryptedMessage === false) {
    echo "Encryption failed.\n";
    exit;
}

// Збереження зашифрованого повідомлення у файл
file_put_contents('aes_encrypted_message.bin', $encryptedMessage);
echo "Encrypted message saved successfully.\n";

File: bsk_aes_python/generate.py
from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes
import binascii
import time

# Wiadomość do zaszyfrowania
message = "Hello, this is a secret message! Don't tell anyone that you got it, You must help Me. Right now I'm sitting in the prison"

# Generowanie losowego 128-bitowego klucza (16 bajtów)
key = get_random_bytes(16)
print("Generated Key (hex):", binascii.hexlify(key).decode())

# Zapisanie klucza do pliku
with open('aes_key.bin', 'wb') as key_file:
    key_file.write(key)

# Dopasowanie wiadomości do wielokrotności 16 bajtów
padded_message = message.ljust(16 * ((len(message) + 15) // 16)).encode()

# Inicjalizacja szyfrowania AES w trybie ECB
cipher = AES.new(key, AES.MODE_ECB)
encrypted_message = cipher.encrypt(padded_message)

# Zapisanie zaszyfrowanej wiadomości do pliku
with open('aes_encrypted_message.bin', 'wb') as encrypted_file:
    encrypted_file.write(encrypted_message)

print("Encrypted message saved successfully.")